src/modules/rewards/components/ActivityButton.tsx

```diff
@@ -39,6 +39,7 @@ const ActivityButton: FC<ActivityButtonProps> = ({
           userId={userId}
           activityTypeId={activityTypeId}
           refetchActivity={refetchActivity}
+          setErrorMessage={setErrorMessage}
```

Comment:
Same default export in this component as well.

Reply:
@abhishek-01k why are we doing this in the first place.

 ```
 let componentToRender;

 default:
      componentToRender = (
        <DefaultActivityButton
          userId={userId}
          activityTypeId={activityTypeId}
          refetchActivity={refetchActivity}
        />
      );
      break;
  }
  return <div>{componentToRender}</div>;
```

why not directly do

```
  switch (activityType) {
    case 'follow_push_on_discord':
      return 
        <DiscordActivityButton
          userId={userId}
          activityTypeId={activityTypeId}
          refetchActivity={refetchActivity}
          setErrorMessage={setErrorMessage}
        />
```

Reply:
We also don't need `break` everywhere. this can be removed after we use my approach.